# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1251018736 -7200
#      Sun Aug 23 11:12:16 2009 +0200
# Branch release-3-2-x
# Node ID 634697062fe15adbb8659fd850e5ffb7335ebc9f
# Parent  899122410ea591b3adca570973d0d4166dd0a3a6
acx_blas_f77_func.m4 (ACX_BLAS_F77_FUNC): Save and restore LIBS

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-18  John W. Eaton  <jwe@octave.org>
+
+	* acx_blas_f77_func.m4 (ACX_BLAS_F77_FUNC): Save and restore LIBS.
+
 2009-07-16  John W. Eaton  <jwe@octave.org>
 
 	* Makeconf.in (MAGICK_LIBS): Don't pass --ldflags to $(MAGICK_CONFIG).
 
 2009-07-09  John W. Eaton  <jwe@octave.org>
 
 	* configure.in: Don't use system strftime on MinGW systems.
 
diff --git a/acx_blas_f77_func.m4 b/acx_blas_f77_func.m4
--- a/acx_blas_f77_func.m4
+++ b/acx_blas_f77_func.m4
@@ -59,16 +59,17 @@
 AC_DEFUN([ACX_BLAS_F77_FUNC], [
 AC_PREREQ(2.50)
 AC_REQUIRE([ACX_BLAS])
 
 # F77 call-compatibility checks
 if test "$cross_compiling" = yes ; then
 	ifelse($3, ,$1,$3)
 elif test x"$acx_blas_ok" = xyes; then
+	save_acx_blas_f77_func_LIBS="$LIBS"
 	LIBS="$BLAS_LIBS $LIBS"
 	AC_LANG_PUSH(Fortran 77)
 # LSAME check (LOGICAL return values)
 	AC_MSG_CHECKING([whether LSAME is called correctly from Fortran])
 	AC_RUN_IFELSE(AC_LANG_PROGRAM(,[[
       logical lsame,w
       external lsame
       character c1,c2
@@ -152,16 +153,17 @@ elif test x"$acx_blas_ok" = xyes; then
 		-a $acx_blas_cdotu_fcall_ok = yes \
 		-a $acx_blas_zdotu_fcall_ok = yes ; then
 		acx_blas_f77_func_ok=yes;
 		$1
 	else
 		acx_blas_f77_func_ok=no;
 		$2
 	fi
+	LIBS="$save_acx_blas_f77_func_LIBS"
 fi
 
 ])dnl ACX_BLAS_F77_FUNC
 
 AC_DEFUN([ACX_BLAS_WITH_F77_FUNC], [
 AC_PREREQ(2.50)
 ACX_BLAS([# disable special action], [])
 if test x$acx_blas_ok = xyes ; then
